src/main/java/uk/gov/hmcts/dts/fact/services/CourtService.java

```diff
@@ -171,6 +171,25 @@ public ServiceAreaWithCourtReferencesWithDistance getNearestCourtsByPostcodeSear
         return new ServiceAreaWithCourtReferencesWithDistance(serviceArea, convert(courts));
     }
 
+    public ServiceAreaWithCourtReferencesWithDistance getNearestCourtsByPostcodeActionAndAreaOfLawSearch(final String postcode, final String serviceAreaSlug, final String action) {
+
+        final Optional<ServiceArea> serviceAreaOptional = serviceAreaRepository.findBySlugIgnoreCase(serviceAreaSlug);
+        final Optional<MapitData> optionalMapitData = mapitService.getMapitData(postcode);
+
+        if (serviceAreaOptional.isEmpty() || optionalMapitData.isEmpty()) {
```

Comment:
I think it's good to check that we have a service area found and that the mapit data exists 

It might be better for us to raise a 404 not found exception if they are not present however? I think otherwise the resulting object will only contain the service area name (the models constructor takes a slug, not a service area too), so it would not be helpful or accurate

So something like raise 404 would suffice I believe